File: protocol/src/test/ruby/zfooruby/packet/SimpleObject.rb

class SimpleObject
    attr_accessor :c  # int
    attr_accessor :g  # bool
    def initialize()
        @c = 0
        @g = false
    end
end

class SimpleObjectRegistration
    def protocolId()
        return 104
    end

    def write(buffer, packet)
        if packet.nil?
            buffer.writeInt(0)
            return
        end
        buffer.writeInt(-1)
        buffer.writeInt(packet.c)
        buffer.writeBool(packet.g)
    end

    def read(buffer)
        length = buffer.readInt()
        if length == 0
            return nil
        end
        beforeReadIndex = buffer.getReadOffset()
        packet = SimpleObject.new()
        result0 = buffer.readInt()
        packet.c = result0
        result1 = buffer.readBool() 
        packet.g = result1
        if length > 0
            buffer.setReadOffset(beforeReadIndex + length)
        end
        return packet
    end
end

File: protocol/src/test/swift/zfooswift/packet/SimpleObject.swift
import Foundation


class SimpleObject : IProtocol {
    var c: Int = 0
    var g: Bool = false
    
    func protocolId() -> Int {
        return 104
    }
}

class SimpleObjectRegistration : IProtocolRegistration {
    func write(_ buffer: ByteBuffer, _ packet: Any?) {
        if (packet == nil) {
            buffer.writeInt(0)
            return
        }
        let message = packet as! SimpleObject
        buffer.writeInt(-1)
        buffer.writeInt(message.c)
        buffer.writeBool(message.g)
    }
    
    func read(_ buffer: ByteBuffer) -> Any {
        let length = buffer.readInt()
        let packet = SimpleObject()
        if (length == 0) {
            return packet
        }
        let beforeReadIndex = buffer.getReadOffset()
        let result0 = buffer.readInt()
        packet.c = result0
        let result1 = buffer.readBool()
        packet.g = result1
        if (length > 0) {
            buffer.setReadOffset(beforeReadIndex + length)
        }
        return packet
    }
}